client/app/src/main/java/com/example/criengine/Activities/RootActivity.java

```diff
@@ -131,6 +130,21 @@ public Fragment createFragment(int position) {
 
             return null;
         }
+
+        @Override
+        public int getItemViewType(int position) {
+            return super.getItemViewType(position);
+        }
+    }
+
+
+    /**
+     * Reloads the activity on the current page
+     */
+    public void refresh() {
+        Intent intent = new Intent(this, RootActivity.class);
+        intent.putExtra("Index", PAGE.values()[viewPager.getCurrentItem()]);
+        startActivity(intent);

```

Comment:
This method cannot be used on page change because simply calling refresh() in the `onPageChange` causes a crash since: onPageChangeCallback -> refresh -> restart activity -> onPageChangeCallback  -> refresh ...